tests/debugger/utils.py

```diff
@@ -71,6 +71,15 @@ class _Base_Debugger_Test:
     rc_state = None
     weblog_responses = []
 
+    def initialize_weblog_remote_config(self):
+        if self.get_tracer()["language"] == "ruby":
+            # Ruby tracer initializes remote configuration client from
+            # middleware that is only invoked during request processing.
+            # Therefore, we need to issue a request to the application for
+            # remote config to start.
+            response = weblog.get("/debugger/init")
+            assert response.status_code == 200
```

Comment:
Sorry, I also miss that point. `setup` function must never fail. If you want to add any assertion, it must be done inside test functions.

```
def setup_stuff:
    self.init_response = weblog.get("/debugger/init")

def test_stuff:
    assert self.init_response.status_code == 200
```

Reply:
Why must `setup` never fail?

Reply:
"Test functions" are instrumented by the framework to offer a friendly output on failure (logs, some variables values on `assert`. 

`setup` functions are executed during the lifespan of the tested infra, but they are not "test functions". If they fails, you'll have a basic stack trace painful to understand